app.ts

```diff
@@ -89,24 +88,25 @@ app.use(passport.initialize());
 app.use(passport.session());
 
 passport.use(
-  new LocalStrategy((username, password, done) => {
+  new LocalStrategy(async (username, password, done) => {
     let _user;
-    User.findByUsername(username)
+    const success = await User.findByUsername(username)
       .then((user) => {
         if (!user) return false;
         _user = user;
         return user.authenticate(password);
       })
-      .then((result) => result && _user)
-      .nodeify(done);
+      .then((result) => result && _user);
+    done(success);
```

Comment:
You could consider using async/await without .then().

Eg. [es7 async await](https://medium.com/@habibridho/javascript-es7-async-await-tutorial-64275c81ce2e)